packages/tools/src/utilities/segmentation/contourAndFindLargestBidirectional.ts

```diff
@@ -0,0 +1,294 @@
+import { cache } from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { vec3 } from 'gl-matrix';
+
+import { generateContourSetsFromLabelmap } from '../rtstruct';
+import SegmentationRepresentations from '../../enums/SegmentationRepresentations';
+
+export type BidirectionalData = {
+  majorAxis: [Types.Point3, Types.Point3];
+  minorAxis: [Types.Point3, Types.Point3];
+  maxMajor: number;
+  maxMinor: number;
+  segmentIndex: number;
+  label?: string;
+  color?: string | number[];
+  referencedImageId: string;
+  FrameOfReferenceUID: string;
+};
+
+const EPSILON = 1e-2;
+const { Labelmap } = SegmentationRepresentations;
+
+/**
+ * Generates a contour object over the segment, and then uses the contouring to
+ * find the largest bidirectional object that can be applied within the acquisition
+ * plane that is within the segment index, or the contained segment indices.
+ *
+ * @param segmentations.segments - a list of segments to apply the contour to.
+ * @param segmentations.segments.containedSegmentIndices - a set of segment indexes equivalent to the primary segment
+ * @param segmentations.segments.label - the label for the segment
+ * @param segmentations.segments.color - the color to use for the segment label
+ */
```

Comment:
I don't understand why this is plural (segmentations), does it work on multiple segmentations at the same time (I don't think based on the code I see), so lets rename it